src/azure-cli-testsdk/azure/cli/testsdk/base.py

```diff
@@ -70,32 +70,49 @@ def __init__(self, method_name, config_file=None, recording_name=None,
         self.cli_ctx = TestCli()
         self.name_replacer = GeneralNameReplacer()
         self.kwargs = {}
+
+        default_recording_processors = [
+            SubscriptionRecordingProcessor(MOCKED_SUBSCRIPTION_ID),
+            OAuthRequestResponsesFilter(),
+            LargeRequestBodyProcessor(),
+            LargeResponseBodyProcessor(),
+            DeploymentNameReplacer(),
+            RequestUrlNormalizer(),
+            self.name_replacer
+        ]
+
+        default_replay_processors = [
+            LargeResponseBodyReplacer(),
+            DeploymentNameReplacer(),
+            RequestUrlNormalizer(),
+        ]
+
+        default_recording_patches = [patch_main_exception_handler]
+
+        default_replay_patches = [
+            patch_main_exception_handler,
+            patch_time_sleep_api,
+            patch_long_run_operation_delay,
+            patch_load_cached_subscriptions,
+            patch_retrieve_token_for_user,
+            patch_progress_controller,
+        ]
+
+        def _merge_lists(base, patches):
+            merged = list(base)
+            if patches and not isinstance(patches, list):
+                patches = [patches]
+            if patches:
+                merged = list(set(merged).union(set(patches)))
+            return merged
+
         super(ScenarioTest, self).__init__(
             method_name,
             config_file=config_file,
-            recording_processors=recording_processors or [
-                SubscriptionRecordingProcessor(MOCKED_SUBSCRIPTION_ID),
-                OAuthRequestResponsesFilter(),
-                LargeRequestBodyProcessor(),
-                LargeResponseBodyProcessor(),
-                DeploymentNameReplacer(),
-                RequestUrlNormalizer(),
-                self.name_replacer,
-            ],
-            replay_processors=replay_processors or [
-                LargeResponseBodyReplacer(),
-                DeploymentNameReplacer(),
-                RequestUrlNormalizer(),
-            ],
-            recording_patches=recording_patches or [patch_main_exception_handler],
-            replay_patches=replay_patches or [
-                patch_main_exception_handler,
-                patch_time_sleep_api,
-                patch_long_run_operation_delay,
-                patch_load_cached_subscriptions,
-                patch_retrieve_token_for_user,
-                patch_progress_controller,
-            ],
+            recording_processors=_merge_lists(default_recording_processors, recording_processors),
+            replay_processors=_merge_lists(default_replay_processors, replay_processors),
+            recording_patches=_merge_lists(default_recording_patches, recording_patches),
+            replay_patches=_merge_lists(default_replay_patches, replay_patches),
```

Comment:
Prev. is was `or`. Now you are merging.
Is this intended and do we want to support merging?

Reply:
Yes, this was very intended. The only reason is was "or" for so long was because no one ever tried to use it... we just always added any processors we needed to the core testbase (which isn't really a good idea...)